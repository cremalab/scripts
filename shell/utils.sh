#!/bin/bash
function print {
  echo ""
  echo "-----------------------------------------------------"
  echo $1
  echo "-----------------------------------------------------"
}