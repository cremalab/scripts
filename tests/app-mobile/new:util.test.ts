import { printFileContents } from "../utils/printFileContents"
import { cleanup } from "../utils/cleanup"
import { generateNewUtil } from "../utils/generateNewUtil"
import { navigateToConsumer } from "../utils/navigateToConsumer"
import { printTreeOfNew } from "../utils/printTreeOfNew"
import { validateTypeScript } from "../utils/validateTypeScript"
import { validateUnitTests } from "../utils/validateUnitTests"

beforeAll(navigateToConsumer("app-mobile"))

describe("new:util", () => {
  beforeAll(generateNewUtil)

  afterAll(() => cleanup("./src/utils/foo"))

  it("generates directory with expected files", () => {
    const tree = printTreeOfNew("utils", "foo")
    expect(tree).toMatchInlineSnapshot(`
    "foo
    ├── foo.test.ts
    ├── foo.ts
    └── index.ts
    "
    `)
  })

  it("TypeScript compiles", async () => {
    const result = await validateTypeScript()
    expect(result).toEqual("")
  })

  it("generates directory with expected `index.ts` contents", async () => {
    const fileContents = await printFileContents("./src/utils/foo/index.ts")
    expect(fileContents).toMatchSnapshot()
  })

  it("generates directory with expected `foo.ts` contents", async () => {
    const fileContents = await printFileContents("./src/utils/foo/foo.ts")
    expect(fileContents).toMatchSnapshot()
  })

  it("generates directory with expected `test.ts` contents", async () => {
    const fileContents = await printFileContents("./src/utils/foo/foo.test.ts")
    expect(fileContents).toMatchSnapshot()
  })

  it("generates working tests", async () => {
    const result = await validateUnitTests("./src/utils/foo/foo.test.ts")
    expect(result).toEqual("PASS ./src/utils/foo/foo.test.ts")
  })
})
