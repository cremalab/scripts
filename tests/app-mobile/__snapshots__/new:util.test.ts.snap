// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new:util generates directory with expected \`foo.ts\` contents 1`] = `
"/**
 * foo
 * ------------------------------------------------
 * DESCRIPTION_HERE
 */

export const foo = <A>(x: A) => x
"
`;

exports[`new:util generates directory with expected \`index.ts\` contents 1`] = `
"export { foo } from \\"./foo\\"
"
`;

exports[`new:util generates directory with expected \`test.ts\` contents 1`] = `
"import { foo } from \\"./foo\\"

describe(\\"foo\\", () => {
  it(\\"works\\", () => {
    // Arrange
    const valA = 1

    // Act
    const received = foo(valA)
    const expected = 1

    // Assert
    expect(received).toEqual(expected)
  })
})
"
`;
